concert/coroutines/sinks.py

```diff
@@ -59,23 +63,36 @@ def __call__(self, producer):
 
         Coroutine interface for processing in a pipeline.
         """
+        if self.reset_on_call:
+            self.reset()
+
         if isinstance(self.items, list):
             return self._process(producer)
         else:
             return self._process_numpy(producer)
 
+    def reset(self):
+        if isinstance(self.items, list):
+            del self.items[:]
+        else:
+            self.items = np.empty((0,) + self._shape[1:], dtype=self._dtype)
```

Comment:
`self.items = np.empty_like(self.items)` should do same, or?

Reply:
No, self.items can grow and you want to reset to the shape given in contructor.

Reply:
self.items = np.empty_like(self.items[0]) ?